File: lib/js/src/Aeson.js
'use strict';

var Block = require("bs-platform/lib/js/block.js");
var Aeson_decode = require("./Aeson_decode.js");
var Aeson_encode = require("./Aeson_encode.js");
var Caml_js_exceptions = require("bs-platform/lib/js/caml_js_exceptions.js");

function decode(json) {
  var v;
  try {
    v = Aeson_decode.bool(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.Bool.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

function encode(prim) {
  return prim;
}

var Bool = {
  decode: decode,
  decodeUnsafe: Aeson_decode.bool,
  encode: encode
};

function decode$1(json) {
  var v;
  try {
    v = Aeson_decode.bigint(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.Bigint.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

var Bigint = {
  decode: decode$1,
  decodeUnsafe: Aeson_decode.bigint,
  encode: Aeson_encode.bigint
};

function decode$2(json) {
  var v;
  try {
    v = Aeson_decode.$$float(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.Float.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

function encode$1(prim) {
  return prim;
}

var Float = {
  decode: decode$2,
  decodeUnsafe: Aeson_decode.$$float,
  encode: encode$1
};

function decode$3(json) {
  var v;
  try {
    v = Aeson_decode.$$int(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.Int.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

function encode$2(prim) {
  return prim;
}

var Int = {
  decode: decode$3,
  decodeUnsafe: Aeson_decode.$$int,
  encode: encode$2
};

function decode$4(json) {
  var v;
  try {
    v = Aeson_decode.int32(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.Int32.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

function encode$3(prim) {
  return prim;
}

var Int32 = {
  decode: decode$4,
  decodeUnsafe: Aeson_decode.int32,
  encode: encode$3
};

function decode$5(json) {
  var v;
  try {
    v = Aeson_decode.nativeint(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.Nativeint.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

function encode$4(prim) {
  return prim;
}

var Nativeint = {
  decode: decode$5,
  decodeUnsafe: Aeson_decode.nativeint,
  encode: encode$4
};

function decode$6(json) {
  var v;
  try {
    v = Aeson_decode.uint8(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.UInt8.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

var UInt8 = {
  decode: decode$6,
  decodeUnsafe: Aeson_decode.uint8,
  encode: Aeson_encode.uint8
};

function decode$7(json) {
  var v;
  try {
    v = Aeson_decode.uint16(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.UInt16.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

var UInt16 = {
  decode: decode$7,
  decodeUnsafe: Aeson_decode.uint16,
  encode: Aeson_encode.uint16
};

function decode$8(json) {
  var v;
  try {
    v = Aeson_decode.uint32(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.UInt32.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

var UInt32 = {
  decode: decode$8,
  decodeUnsafe: Aeson_decode.uint32,
  encode: Aeson_encode.uint32
};

function decode$9(json) {
  var v;
  try {
    v = Aeson_decode.uint64(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.UInt64.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

var UInt64 = {
  decode: decode$9,
  decodeUnsafe: Aeson_decode.uint64,
  encode: Aeson_encode.uint64
};

function decode$10(json) {
  var v;
  try {
    v = Aeson_decode.string(json);
  }
  catch (raw_exn){
    var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["Aeson.String.decode: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  return /* Ok */Block.__(0, [v]);
}

function encode$5(prim) {
  return prim;
}

var $$String = {
  decode: decode$10,
  decodeUnsafe: Aeson_decode.string,
  encode: encode$5
};

var Decode = /* alias */0;

var Encode = /* alias */0;

var Compatibility = /* alias */0;

exports.Decode = Decode;
exports.Encode = Encode;
exports.Compatibility = Compatibility;
exports.Bool = Bool;
exports.Bigint = Bigint;
exports.Float = Float;
exports.Int = Int;
exports.Int32 = Int32;
exports.Nativeint = Nativeint;
exports.UInt8 = UInt8;
exports.UInt16 = UInt16;
exports.UInt32 = UInt32;
exports.UInt64 = UInt64;
exports.$$String = $$String;
/* Aeson_decode Not a pure module */

File: src/Aeson.ml
module Decode = Aeson_decode
module Encode = Aeson_encode
module Compatibility = Aeson_compatibility

module type DecodeUnsafe = sig
  type t
  val decodeUnsafe : Js.Json.t -> t
end

module type DecodeSafe = sig
  type t
  val decode : Js.Json.t -> (t, string) Belt.Result.t
  include DecodeUnsafe with type t := t
end

module type Encode = sig
  type t
  val encode : t -> Js.Json.t
end

module type EncodeAndDecode = sig
  include DecodeSafe
  include Encode with type t := t
end

module Bool : EncodeAndDecode with type t = bool = struct
  type t = bool

  let decodeUnsafe = Aeson_decode.bool
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.Bool.decode: " ^ err)
    
  let encode = Aeson_encode.bool
end

module Bigint : EncodeAndDecode with type t = Bigint.t = struct
  type t = Bigint.t

  let decodeUnsafe = Aeson_decode.bigint
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.Bigint.decode: " ^ err)
    
  let encode = Aeson_encode.bigint
end
                                                 
module Float : EncodeAndDecode with type t = float = struct
  type t = float

  let decodeUnsafe = Aeson_decode.float
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.Float.decode: " ^ err)
    
  let encode = Aeson_encode.float
end

module Int : EncodeAndDecode with type t = int = struct
  type t = int

  let decodeUnsafe = Aeson_decode.int
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.Int.decode: " ^ err)
    
  let encode = Aeson_encode.int
end

module Int32 : EncodeAndDecode with type t = int32 = struct
  type t = int32

  let decodeUnsafe = Aeson_decode.int32
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.Int32.decode: " ^ err)
    
  let encode = Aeson_encode.int32
end

module Nativeint : EncodeAndDecode with type t = nativeint = struct
  type t = nativeint

  let decodeUnsafe = Aeson_decode.nativeint
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.Nativeint.decode: " ^ err)
    
  let encode = Aeson_encode.nativeint
end

module UInt8 : EncodeAndDecode with type t = U.UInt8.t = struct
  type t = U.UInt8.t

  let decodeUnsafe = Aeson_decode.uint8
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.UInt8.decode: " ^ err)
    
  let encode = Aeson_encode.uint8
end

module UInt16 : EncodeAndDecode with type t = U.UInt16.t = struct
  type t = U.UInt16.t

  let decodeUnsafe = Aeson_decode.uint16
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.UInt16.decode: " ^ err)
    
  let encode = Aeson_encode.uint16
end

module UInt32 : EncodeAndDecode with type t = U.UInt32.t = struct
  type t = U.UInt32.t

  let decodeUnsafe = Aeson_decode.uint32
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.UInt32.decode: " ^ err)
    
  let encode = Aeson_encode.uint32
end

module UInt64 : EncodeAndDecode with type t = U.UInt64.t = struct
  type t = U.UInt64.t

  let decodeUnsafe = Aeson_decode.uint64
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.UInt64.decode: " ^ err)
    
  let encode = Aeson_encode.uint64
end
                                                         
module String : EncodeAndDecode with type t = string = struct
  type t = string

  let decodeUnsafe = Aeson_decode.string
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.String.decode: " ^ err)
    
  let encode = Aeson_encode.string
end

module Date : EncodeAndDecode with type t = Js_date.t = struct
  type t = Js_date.t

  let decodeUnsafe = Aeson_decode.date
  let decode = fun json ->
    match (decodeUnsafe json) with
    | v -> Belt.Result.Ok v
    | exception Aeson_decode.DecodeError err -> Belt.Result.Error ("Aeson.Date.decode: " ^ err)
    
  let encode = Aeson_encode.date
end

